osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/utils/DlmsHelper.java

```diff
@@ -1035,4 +1041,53 @@ private void logAndThrowExceptionForUnexpectedResultData(
             + ", value type: "
             + resultDataType);
   }
+
+  public String getScalerUnitValue(final DlmsConnectionManager conn, final CosemObject object)
+      throws ProtocolAdapterException {
+    final int attributeId;
+    if (object.getClassId() == InterfaceClass.REGISTER.id()) {
+      attributeId = RegisterAttribute.SCALER_UNIT.attributeId();
+    } else if (object.getClassId() == InterfaceClass.EXTENDED_REGISTER.id()) {
+      attributeId = ExtendedRegisterAttribute.SCALER_UNIT.attributeId();
+    } else {
+      return null;
+    }
+
+    final Attribute attribute = object.getAttribute(attributeId);
+    if (attribute.getValuetype() == ValueType.FIXED_IN_PROFILE) {
+      return attribute.getValue();
+    } else {
+      final AttributeAddress attributeAddress =
+          new AttributeAddress(object.getClassId(), object.getObis(), attributeId);
+
+      try {
+        final DataObject scalerUnitObject = this.getAttributeValue(conn, attributeAddress);
```

Comment:
The changes are made for Instananeous voltage. For polyphase meters, this approach would result in 3 additional requests to the meter. It would be better to group the requests in one single call (using with_list).
Other approach could be to use the modeltype to determine the scaler_unit, without request to the meter. This approach is under development at the moment for periodic meter readings.